tests/utils.py

```diff
@@ -59,3 +61,12 @@ def point_has_coordinates_types(point: Point,
                                 y_type: Type[Coordinate]) -> bool:
     x, y = point
     return isinstance(x, x_type) and isinstance(y, y_type)
+
+
+def has_no_consecutive_repetitions(iterable: Iterable[Domain]) -> bool:
+    return any(capacity(group) == 1
+               for _, group in groupby(iterable))
+
+
+def capacity(iterable: Iterable[Domain]) -> int:
+    return sum(1 for _ in iterable)
```

Comment:
This function is also in your `lz` library. Don't you want to use it?

Reply:
I don't want to include non-lightweight dependency for a single function, we are not in javascript afterall